File: benchmarks/Python/quicksort.py
def quicksort(a, lo, hi):
    i = lo
    j = hi
    while i < hi:
        pivot = a[(lo + hi) // 2]
        while i <= j:
            while a[i] < pivot:
                i += 1

            while a[j] > pivot:
                j -= 1

            if i <= j:
                a[i], a[j] = a[j], a[i]
                i += 1
                j -= 1

        if lo < j:
            quicksort(a, lo, j)

        lo = i
        j = hi
    return a

File: benchmarks/C/src/quicksort.c
void quicksort(int *A, int lo, int hi) {
    int i = lo;
    int j = hi;
    while (i < hi) {
        int pivot = A[(lo + hi) / 2];
        while (i <= j) {
            while (A[i] < pivot) {
                i = i + 1;
            }

            while (A[j] > pivot) {
                j = j - 1;
            }

            if (i <= j) {
                int t = A[i];
                A[i] = A[j];
                A[j] = t;
                i = i + 1;
                j = j - 1;
            }
        }

        if (lo < j) {
            quicksort(A, lo, j);
        }

        lo = i;
        j = hi;
    }
}